gc/base/Configuration.cpp

```diff
@@ -521,31 +521,26 @@ MM_Configuration::createParallelDispatcher(MM_EnvironmentBase *env, omrsig_handl
 }
 
 #if defined(J9VM_OPT_CRIU_SUPPORT)
-void
-MM_Configuration::adjustGCThreadCountForCheckpoint(MM_EnvironmentBase* env)
-{
-	MM_GCExtensionsBase* extensions = env->getExtensions();
-	MM_ParallelDispatcher* dispatcher = extensions->dispatcher;
-
-	dispatcher->contractThreadPool(env, extensions->checkpointGCthreadCount);
-}
-
 bool
-MM_Configuration::reinitializeGCThreadCountForRestore(MM_EnvironmentBase* env)
+MM_Configuration::reinitializeForRestore(MM_EnvironmentBase* env)
 {
 	MM_GCExtensionsBase* extensions = env->getExtensions();
 
-	uintptr_t checkpointThreadCount = extensions->dispatcher->threadCountMaximum();
+	/* This path should not be taken by balanced GC, memory Pool iteration can be expensive with many pools.
+	 * Currently, balanced is unsupported with CRIU. In the future, ConfigurationIncrementalGenerational
+	 * may need to override this method.
+	 */
+	Assert_MM_true(!extensions->isVLHGC());
```

Comment:
A general comment: isVLHGC() is a hack that should not exist (but rather virtual dispatch should be used) especially in OMR (that should not be even aware of VLHGC/Balanced). We should try to minimize it's usage and even try to get rid of some existing usages (unfortunately it's already very wide spread). 

So, I don't think this assert and any comments referring to Balanced and ConfigurationIncrementalGenerational is needed here, but if you think this will be removed in not too distant future, I'm ok to leave it...